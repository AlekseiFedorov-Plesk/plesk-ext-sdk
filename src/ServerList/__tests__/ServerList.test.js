// Copyright 1999-2018. Plesk International GmbH. All rights reserved.

import { createElement, Pagination, List } from '@plesk/ui-library';
import { ServerList } from '../ServerList';
import { shallow } from 'enzyme';

const generateData = length => [...Array(length)].map((_, index) => ({ key: String(index), column1: index }));

describe('ServerList', () => {
    it('renders correctly', () => {
        const data = generateData(5);
        const promise = Promise.resolve({ data, totalItems: data.length });
        const get = jest.fn(() => promise);

        const wrapper = shallow(
            <ServerList
                config={{
                    baseUrl: '',
                }}
                api={{
                    get,
                }}
                columns={[{
                    key: 'column1',
                    title: 'Column1',
                }]}
                action="/api/test"
            />
        );

        expect.assertions(4);

        return promise.then(() => {
            expect(get).toHaveBeenCalledWith('/api/test', { pageSize: 25, pageNumber: 1 });
            expect(wrapper.state()).toMatchObject({
                pageNumber: 1,
                totalPages: 1,
                sortColumn: null,
                sortDirection: 'ASC',
            });
            expect(wrapper.state().data).toEqual(data);
            expect(wrapper).toMatchSnapshot();
        });
    });

    it('render pagination correctly', () => {
        const data = generateData(60);
        const promise = Promise.resolve({ data: data.slice(0, 25), totalItems: data.length });
        const get = jest.fn(() => promise);

        const wrapper = shallow(
            <ServerList
                config={{
                    baseUrl: '',
                }}
                api={{
                    get,
                }}
                columns={[{
                    key: 'column1',
                    title: 'Column1',
                }]}
                action="/api/test"
            />
        );

        expect.assertions(5);

        return promise
            .then(() => {
                expect(get).toHaveBeenCalledWith('/api/test', { pageSize: 25, pageNumber: 1 });
                expect(wrapper.state()).toMatchObject({
                    pageNumber: 1,
                    totalPages: 3,
                });
                expect(wrapper).toMatchSnapshot();

                wrapper.find(Pagination).simulate('select', 2);
                expect(get).toHaveBeenCalledWith('/api/test', { pageSize: 25, pageNumber: 2 });
                expect(wrapper.state()).toMatchObject({
                    pageNumber: 2,
                    totalPages: 3,
                });
            });
    });

    it('render sort correctly', () => {
        const data = generateData(60);
        const promise = Promise.resolve({ data: data.slice(0, 25), totalItems: data.length });
        const get = jest.fn(() => promise);

        const wrapper = shallow(
            <ServerList
                config={{
                    baseUrl: '',
                }}
                api={{
                    get,
                }}
                columns={[{
                    key: 'column1',
                    title: 'Column1',
                }]}
                action="/api/test"
                defaultSortColumn="column1"
            />
        );

        expect.assertions(5);

        return promise.then(() => {
            expect(get).toHaveBeenCalledWith('/api/test', {
                pageSize: 25,
                pageNumber: 1,
                sortColumn: 'column1',
                sortDirection: 'ASC',
            });
            expect(wrapper.state()).toMatchObject({
                pageNumber: 1,
                totalPages: 3,
                sortColumn: 'column1',
                sortDirection: 'ASC',
            });
            expect(wrapper).toMatchSnapshot();

            wrapper.find(List).simulate('sortChange', { sortColumn: 'column1', sortDirection: 'DESC' });
            expect(get).toHaveBeenCalledWith('/api/test', {
                pageSize: 25,
                pageNumber: 1,
                sortColumn: 'column1',
                sortDirection: 'DESC',
            });
            expect(wrapper.state()).toMatchObject({
                sortColumn: 'column1',
                sortDirection: 'DESC',
            });
        });
    });
});
