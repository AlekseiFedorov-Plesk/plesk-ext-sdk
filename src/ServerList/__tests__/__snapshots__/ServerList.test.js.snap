// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServerList renders correctly 1`] = `
<div>
  <List
    baseClassName="pul-list"
    className={null}
    columns={
      Array [
        Object {
          "key": "column1",
          "title": "Column1",
        },
      ]
    }
    data={
      Array [
        Object {
          "column1": 0,
          "key": "0",
        },
        Object {
          "column1": 1,
          "key": "1",
        },
        Object {
          "column1": 2,
          "key": "2",
        },
        Object {
          "column1": 3,
          "key": "3",
        },
        Object {
          "column1": 4,
          "key": "4",
        },
      ]
    }
    onSelectionChange={null}
    onSortChange={[Function]}
    rowKey="key"
    selection={Array []}
    sortColumn={null}
    sortDirection="ASC"
  />
</div>
`;

exports[`ServerList renders errors correctly 1`] = `
<div>
  <List
    baseClassName="pul-list"
    className={null}
    columns={
      Array [
        Object {
          "key": "column1",
          "title": "Column1",
        },
      ]
    }
    data={Array []}
    onSelectionChange={null}
    onSortChange={[Function]}
    rowKey="key"
    selection={Array []}
    sortColumn={null}
    sortDirection="ASC"
  />
</div>
`;

exports[`ServerList renders pagination correctly 1`] = `
<div>
  <List
    baseClassName="pul-list"
    className={null}
    columns={
      Array [
        Object {
          "key": "column1",
          "title": "Column1",
        },
      ]
    }
    data={
      Array [
        Object {
          "column1": 0,
          "key": "0",
        },
        Object {
          "column1": 1,
          "key": "1",
        },
        Object {
          "column1": 2,
          "key": "2",
        },
        Object {
          "column1": 3,
          "key": "3",
        },
        Object {
          "column1": 4,
          "key": "4",
        },
        Object {
          "column1": 5,
          "key": "5",
        },
        Object {
          "column1": 6,
          "key": "6",
        },
        Object {
          "column1": 7,
          "key": "7",
        },
        Object {
          "column1": 8,
          "key": "8",
        },
        Object {
          "column1": 9,
          "key": "9",
        },
        Object {
          "column1": 10,
          "key": "10",
        },
        Object {
          "column1": 11,
          "key": "11",
        },
        Object {
          "column1": 12,
          "key": "12",
        },
        Object {
          "column1": 13,
          "key": "13",
        },
        Object {
          "column1": 14,
          "key": "14",
        },
        Object {
          "column1": 15,
          "key": "15",
        },
        Object {
          "column1": 16,
          "key": "16",
        },
        Object {
          "column1": 17,
          "key": "17",
        },
        Object {
          "column1": 18,
          "key": "18",
        },
        Object {
          "column1": 19,
          "key": "19",
        },
        Object {
          "column1": 20,
          "key": "20",
        },
        Object {
          "column1": 21,
          "key": "21",
        },
        Object {
          "column1": 22,
          "key": "22",
        },
        Object {
          "column1": 23,
          "key": "23",
        },
        Object {
          "column1": 24,
          "key": "24",
        },
      ]
    }
    onSelectionChange={null}
    onSortChange={[Function]}
    rowKey="key"
    selection={Array []}
    sortColumn={null}
    sortDirection="ASC"
  />
  <Pagination
    baseClassName="pul-pagination"
    className={null}
    current={1}
    isShowNavigationButtons={true}
    isShowPageButtons={null}
    maxPageButtons={11}
    onSelect={[Function]}
    total={3}
  />
</div>
`;

exports[`ServerList renders sort correctly 1`] = `
<div>
  <List
    baseClassName="pul-list"
    className={null}
    columns={
      Array [
        Object {
          "key": "column1",
          "title": "Column1",
        },
      ]
    }
    data={
      Array [
        Object {
          "column1": 0,
          "key": "0",
        },
        Object {
          "column1": 1,
          "key": "1",
        },
        Object {
          "column1": 2,
          "key": "2",
        },
        Object {
          "column1": 3,
          "key": "3",
        },
        Object {
          "column1": 4,
          "key": "4",
        },
        Object {
          "column1": 5,
          "key": "5",
        },
        Object {
          "column1": 6,
          "key": "6",
        },
        Object {
          "column1": 7,
          "key": "7",
        },
        Object {
          "column1": 8,
          "key": "8",
        },
        Object {
          "column1": 9,
          "key": "9",
        },
        Object {
          "column1": 10,
          "key": "10",
        },
        Object {
          "column1": 11,
          "key": "11",
        },
        Object {
          "column1": 12,
          "key": "12",
        },
        Object {
          "column1": 13,
          "key": "13",
        },
        Object {
          "column1": 14,
          "key": "14",
        },
        Object {
          "column1": 15,
          "key": "15",
        },
        Object {
          "column1": 16,
          "key": "16",
        },
        Object {
          "column1": 17,
          "key": "17",
        },
        Object {
          "column1": 18,
          "key": "18",
        },
        Object {
          "column1": 19,
          "key": "19",
        },
        Object {
          "column1": 20,
          "key": "20",
        },
        Object {
          "column1": 21,
          "key": "21",
        },
        Object {
          "column1": 22,
          "key": "22",
        },
        Object {
          "column1": 23,
          "key": "23",
        },
        Object {
          "column1": 24,
          "key": "24",
        },
      ]
    }
    onSelectionChange={null}
    onSortChange={[Function]}
    rowKey="key"
    selection={Array []}
    sortColumn="column1"
    sortDirection="ASC"
  />
  <Pagination
    baseClassName="pul-pagination"
    className={null}
    current={1}
    isShowNavigationButtons={true}
    isShowPageButtons={null}
    maxPageButtons={11}
    onSelect={[Function]}
    total={3}
  />
</div>
`;
